src/main/java/alpha/FileHandler.java

```diff
@@ -0,0 +1,192 @@
+     */
+    // Saves Task to file whenever a Task is added
+    public void saveToFile(Task task) {
+        try {
+            writer = new FileWriter(alphaTxt, true);
+            if (alphaTxt.length() != 0) {
+                writer.write(System.lineSeparator());
+            }
+            if (task instanceof Deadline) {
+                writer.write("D |" + task.getStatusIcon() + "| " + task.getDescription() + " | "
+                        + ((Deadline) task).getDateBy() + " " + ((Deadline) task).getTimeBy());
+            } else if (task instanceof ToDo) {
+                writer.write("T |" + task.getStatusIcon() + "| " + task.getDescription());
+            } else {
+                writer.write("E |" + task.getStatusIcon() + "| " + task.getDescription() + " | " + ((Event) task).getStart()
+                        + " | " + ((Event) task).getEnd());
+            }
+            writer.close();
+        } catch (IOException e) {
+            System.out.println("I/O Exception");
+        }
+    }
+
+    /**
+     * Checks or unchecks a task in the file depending on the second boolean parameter.
+     *
+     * @param index The index of the task to be marked or unmarked.
+     * @param check Boolean of whether the task is to be checked or unchecked.
+     */
+    // Checks or unchecks a task in the file. Does so by creating a temp file that copies everything over
+    // except the task that is being checked/unchecked.
+    public void checkOrUncheck(int index, boolean check) {
+        try {
+            int temp_index = index;
+            File temp = File.createTempFile("file", ".txt", alphaTxt.getParentFile());
+            writer = new FileWriter(temp, true);
+            reader = new BufferedReader(new FileReader(alphaTxt));
+            String curr = reader.readLine();
+            while (temp_index > 1) {
+                writer.write(curr);
+                writer.write(System.lineSeparator());
+                temp_index--;
+                curr = reader.readLine();
+            }
+            String[] currSplit = curr.split("\\|");
+            if (check) {
+                currSplit[1] = "X";
+            } else {
+                currSplit[1] = " ";
+            }
+            curr = String.join("|", currSplit);
+            writer.write(curr);
+            while ((curr = reader.readLine()) != null) {
+                writer.write(System.lineSeparator());
+                writer.write(curr);
+            }
+
+            reader.close();
+            writer.close();
+            alphaTxt.delete();
+            temp.renameTo(alphaTxt);
+        } catch (IOException e) {
+            System.out.println("There was an issue!");
+        }
+    }
+
+
+    /**
+     * Deletes a task from the file.
+     *
+     * @param index Index of the task to be deleted.
+     */
+    // Deletes a task by creating a temp file and copying everything but the deleted task over.
+    public void delete(int index) {
```

Comment:
same as above regarding the temp file, and perhaps should use a method name that is more specific